zipp/__init__.py

```diff
@@ -313,6 +322,47 @@ def iterdir(self):
         subs = map(self._next, self.root.namelist())
         return filter(self._is_child, subs)
 
+    def match(self, path_pattern):
+        return pathlib.Path(self.at).match(path_pattern)
+
+    def is_symlink(self):
+        return False  # See #82102
+
+    @contextlib.contextmanager
+    def _scandir(self):  # Needed for glob
+        yield self.iterdir()
+
+    def _make_child_relpath(self, relpath):  # Needed for glob
+        return self / relpath
+
+    def glob(self, pattern):
+        if not pattern:
+            raise ValueError("Unacceptable pattern: {!r}".format(pattern))
+        path = pathlib.Path(self.at)
+        drv, root, pattern_parts = path._flavour.parse_parts((pattern,))
+        if drv or root:
+            raise NotImplementedError("Non-relative patterns are unsupported")
```

Comment:
I know this was copied from another implementation, but this `NotImplementedError` is the wrong error. It's not that the behavior is not yet implemented. Instead, this behavior should never be implemented. And in that case, a `ValueError` would be better here.
```suggestion
            raise ValueError("Non-relative patterns are not allowed")
```

Reply:
This concern is bypassed in e587f319826bce03df1dc7233842f054172c1fd2.